File: ModuleDaily/src/main/java/com/qihuan/dailymodule/contract/DailyDetContract.java
package com.qihuan.dailymodule.contract;

import com.qihuan.commonmodule.base.BasePresenter;
import com.qihuan.commonmodule.base.BaseView;
import com.qihuan.dailymodule.model.bean.StoryContentBean;
import com.qihuan.dailymodule.model.bean.StoryExtraBean;

public interface DailyDetContract {
    interface View extends BaseView {
        void storyContent(StoryContentBean storyContent);

        void storyExtra(StoryExtraBean storyExtra);

        void onFavoriteChange(boolean isFavorite);

        void showUpdateFavoriteInfo(boolean isFavorite);
    }

    interface Presenter extends BasePresenter<View> {
        void getStoryContent(int id);

        void getStoryExtra(int id);

        void getFavoriteStory(int id);

        void updateFavoriteStory(int id);
    }
}

File: module_daily/src/main/java/com/qihuan/dailymodule/contract/DailyDetContract.kt
package com.qihuan.dailymodule.contract

import com.qihuan.commonmodule.base.BasePresenter
import com.qihuan.commonmodule.base.BaseView
import com.qihuan.dailymodule.model.bean.StoryContentBean
import com.qihuan.dailymodule.model.bean.StoryExtraBean

interface DailyDetContract {
    interface View : BaseView {
        fun storyContent(storyContent: StoryContentBean)

        fun storyExtra(storyExtra: StoryExtraBean)

        fun onFavoriteChange(isFavorite: Boolean)

        fun showUpdateFavoriteInfo(isFavorite: Boolean)
    }

    interface Presenter : BasePresenter<View> {
        fun getStoryContent(id: Int)

        fun getStoryExtra(id: Int)

        fun getFavoriteStory(id: Int)

        fun updateFavoriteStory(id: Int)
    }
}
